operator/gitops/argocd/pipeline-service/openshift-pipelines/chains-service-monitor.yaml

```diff
@@ -0,0 +1,65 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: chains-metrics-reader
+  namespace: openshift-pipelines
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+---
+apiVersion: v1
+kind: Secret
```

Comment:
Yes, the service account will have default secret created. But by creating and mapping it manually we get a reference to the secret and the JWT from the secret is required in the service monitor to access the metrics. Thanks Sayan for explaining this to me.

Reply:
ok, so that tells me there is something unique for metrics and chains, but I'm still not hearing "why" it is required to access chains metrics @sayan-biswas @avinal .  Is there some separate authentication enabled on the chains metrics service?

Or is this perhaps a gap with say the support behind https://github.com/tektoncd/pipeline/blob/main/config/config-observability.yaml in the main tekton project?  

I'm logged onto the staging cluster as I type, and I don't see similar secrets defined to correspond to the `openshift-pipelines-monitor` ServiceMonitor that the openshift pipeline operator creates for the pipelines controller.

Reply:
ok lengthy discussion with @avinal @sayan-biswas and myself in https://redhat-internal.slack.com/archives/C032EJ007C0/p1692878986840629

we are losing this secret, and the use of bearer token in service monitor